From f9228651b3bbcdddd93d161c33464918d73655af Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Wed, 18 Sep 2019 18:17:23 +0800
Subject: [PATCH] Rule audit_rules_privileged_commands_pam_timestamp_check add
 remediate script

Add remediate script for rule audit_rules_privileged_commands_pam_timestamp_check

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                                | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)
 create mode 100644 linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_pam_timestamp_check/bash/wrlinux1019.sh

diff --git a/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_pam_timestamp_check/bash/wrlinux1019.sh b/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_pam_timestamp_check/bash/wrlinux1019.sh
new file mode 100644
index 0000000..4b33c7a
--- /dev/null
+++ b/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_pam_timestamp_check/bash/wrlinux1019.sh
@@ -0,0 +1,18 @@
+# platform = multi_platform_wrlinux
+
+slash_sbinary="\/usr\/sbin\/pam_timestamp_check"
+sbinary="/usr/sbin/pam_timestamp_check"
+afile="/etc/audit/audit.rules"
+min_auid="1000"
+
+expected_rule="-a always,exit -F path=${sbinary} -F perm=x -F auid>=${min_auid} -F auid!=unset -k privileged"
+
+base_search=$(sed -e '/-a always,exit/!d' -e '/-F path='"${slash_sbinary} "'/!d'          \
+                                -e '/-F path=[^[:space:]]\+/!d'   -e '/-F perm=.*/!d'                                           \
+                                -e '/-F auid>='"${min_auid}"'/!d' -e '/-F auid!=\(4294967295\|unset\)/!d'       \
+                                -e '/-k \|-F key=/!d' "$afile")
+
+
+if [ -z  "$base_search" ]; then
+	echo "${expected_rule}" >>$afile
+fi
-- 
1.9.1

